### PR TITLE
Refactor of HttpUtil and HttpHeaderUtil

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
There currently exists http.HttpUtil, http2.HttpUtil, and http.HttpHeaderUtil. Having 2 HttpUtil methods can be confusing and the utilty methods in the http package could be consolidated.

Modifications:
- Rename http2.HttpUtil to http2.HttpConversionUtil
- Move http.HttpHeaderUtil methods into http.HttpUtil

Result:
Consolidated utilities whose names don't overlap.
Fixes https://github.com/netty/netty/issues/4120

### CODE REVIEW

**Comment by Scottmitch** [comment]
@trustin @nmittler - FYI

**Comment by nmittler** [comment]
LGTM

**Comment by trustin** [comment]
LGTM

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 1]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
                 uri.getHost() == null && uri.getAuthority() == null && uri.getQuery() == null &&
                 uri.getFragment() == null;
     }
+
+    /**
+     * Returns {@code true} if and only if the connection can remain open and
+     * thus 'kept alive'.  This methods respects the value of the
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: remove extra space after the `.`

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 2]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
...
+        if (message.protocolVersion().isKeepAliveDefault()) {
+            return !HttpHeaderValues.CLOSE.contentEqualsIgnoreCase(connection);
+        } else {
+            return HttpHeaderValues.KEEP_ALIVE.contentEqualsIgnoreCase(connection);
+        }
+    }
+
+    /**
+     * Sets the value of the {@code "Connection"} header depending on the
+     * protocol version of the specified message.  This getMethod sets or removes
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: remove extra space after the `.`.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 3]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
...
+            if (keepAlive) {
+                h.set(HttpHeaderNames.CONNECTION, HttpHeaderValues.KEEP_ALIVE);
+            } else {
+                h.remove(HttpHeaderNames.CONNECTION);
+            }
+        }
+    }
+
+    /**
+     * Returns the length of the content.  Please note that this value is
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: remove extra space after the `.`.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 4]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
...
+        if (webSocketContentLength >= 0) {
+            return webSocketContentLength;
+        }
+
+        // Otherwise we don't.
+        throw new NumberFormatException("header not found: " + HttpHeaderNames.CONTENT_LENGTH);
+    }
+
+    /**
+     * Returns the length of the content.  Please note that this value is
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: remove extra space after the `.`.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 5]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
...
+     * @return the content length or {@code defaultValue} if this message does
+     *         not have the {@code "Content-Length"} header or its value is not
+     *         a number. Not to exceed the boundaries of integer.
+     */
+    public static int getContentLength(HttpMessage message, int defaultValue) {
+        return (int) Math.min(Integer.MAX_VALUE, getContentLength(message, (long) defaultValue));
+    }
+
+    /**
+     * Returns the content length of the specified web socket message.  If the
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: remove extra space after the `.`.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 6]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
...
+            return true;
+        }
+
+        // Multiple 'Expect' headers.  Search through them.
+        return message.headers().contains(HttpHeaderNames.EXPECT, HttpHeaderValues.CONTINUE, true);
+    }
+
+    /**
+     * Sets or removes the {@code "Expect: 100-continue"} header to / from the
+     * specified message.  If the specified {@code value} is {@code true},
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: remove extra space after the `.`.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java [hunk 7]
```diff
@@ -42,4 +47,252 @@ public static boolean isAsteriskForm(URI uri) {
...
+     * Checks to see if the transfer encoding in a specified {@link HttpMessage} is chunked
+     *
+     * @param message The message to check
+     * @return True if transfer encoding is chunked, otherwise false
+     */
+    public static boolean isTransferEncodingChunked(HttpMessage message) {
+        return message.headers().contains(HttpHeaderNames.TRANSFER_ENCODING, HttpHeaderValues.CHUNKED, true);
+    }
+
+    public static void setTransferEncodingChunked(HttpMessage m, boolean chunked) {
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: @Scottmitch javadocs for the win ? ;)

**Comment by normanmaurer** [comment]
@Scottmitch just a few comments :) Good work

**Comment by Scottmitch** [comment]
Cherry-picked 4.1 (b6a4f5d) master (fcd8a3e)

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 1 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: Will submit followup commit to fix these comments @normanmaurer. All of this was just copy/paste.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 2 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: done

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 3 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: done

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 4 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: done

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 5 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: done

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 6 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: done

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java (hunk 7 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java]: done

**Comment by Scottmitch** [comment]
@normanmaurer's follow up comments addressed with commit 4.1 (5498fd1) master (52b77a7).  Thanks for review @normanmaurer (and others :smile: )

**Pull request closed without merging**